File: pset1/mario.c
#include <stdio.h>
#include <cs50.h>

void mario(int h);
void left_triangle(int h, int i);
void right_triangle(int h, int i);

int main(void)
{

    int h = -1;

    // Keep prompting the user until they give a valid height
    while (h < 0 || h > 23)
    {
        printf("Height: ");
        h = get_int();
    }
    mario(h);
}

void mario(int h)
{


    for (int i = h; i >= 1; i--)
    {

        left_triangle(h, i);
        printf("  ");
        right_triangle(h, h - i + 1);
        printf("\n");

    }


}

void left_triangle(int h, int i)
{
    for (int j = 1; j <= h; j++)
    {
        if (j < i)
        {
            printf(" ");
        }
        else
        {
            printf("#");
        }
    }
}

void right_triangle(int h, int i)
{
    for (int j = 1; j <= h; j++)
    {
        if (j <= i)
        {
            printf("#");
        }
    }
}

File: pset6/sentimental/mario.py
from cs50 import get_int


def left_triangle(h, i):
    for j in range(1, h + 1):
        if (j < i):
            print(' ', end='')
        else:
            print("#", end='')


def right_triangle(h, i):
    for j in range(1, h + 1):
        if (j <= i):
            print("#", end='')


def get_positive_int(prompt):
    """Prompt user for positive integer"""
    while True:
        n = get_int(prompt)
        if 0 <= n and n < 24:
            break
    return n


h = get_positive_int("Height: ")

for i in range(h, 0, -1):
    left_triangle(h, i)
    print('  ', end='')
    right_triangle(h, h - i + 1)
    print("")